Sources/TuistGenerator/Linter/GraphLinter.swift

```diff
@@ -45,11 +45,39 @@ public class GraphLinter: GraphLinting {
         issues.append(contentsOf: lintMismatchingConfigurations(graphTraverser: graphTraverser))
         issues.append(contentsOf: lintWatchBundleIndentifiers(graphTraverser: graphTraverser))
         issues.append(contentsOf: lintCodeCoverageMode(graphTraverser: graphTraverser))
+        issues.append(contentsOf: lintSchemesUnknownTargets(graphTraverser: graphTraverser))
         return issues
     }
 
     // MARK: - Fileprivate
 
+    private func lintSchemesUnknownTargets(graphTraverser: GraphTraversing) -> [LintingIssue] {
+        let targets = graphTraverser.targets()
+        return graphTraverser.schemes().compactMap { scheme in
+            let unknownTargets = scheme
+                .targetDependencies()
+                .filter { targetReference in
+                    if let target = targets[targetReference.projectPath],
+                       target.keys.contains(targetReference.name)
+                    {
+                        return false
+                    }
+                    return true
+                }
+
+            guard !unknownTargets.isEmpty else { return nil }
+
+            let targetsDescriptionStrings = unknownTargets.map { target in
+                target.name + " " + "(\(target.projectPath.pathString))"
```

Comment:
nit: The absolute path might be too long to print 

```
The following warnings need attention:
 · Cannot find targets UnknownTarget (/Users/.../tuist/fixtures/ios_app_with_custom_scheme/Frameworks/Framework1) defined in Workspace-App
```

we could possibly print a relative path to the workspace

```suggestion
                target.name + " " + "(\(target.projectPath.relative(to: graphTraverser.path).pathString)"
```

which would result in:

```
The following warnings need attention:
 · Cannot find targets UnknownTarget (Frameworks/Framework1)  defined in Workspace-App
```